app/vmagent/remotewrite/remotewrite.go

```diff
@@ -599,41 +676,25 @@ func tryShardingBlockAmongRemoteStorages(rwctxs []*remoteWriteCtx, tssBlock []pr
 			tmpLabels.Labels = hashLabels
 		}
 		h := getLabelsHash(hashLabels)
-		idx := h % uint64(len(shards))
-		i := 0
+
+		// Get the rwctxIdx through consistent hashing and then map it to the index in shards.
+		// The rwctxIdx is not always equal to the shardIdx, for example, when some rwctx are not available.
+		rwctxIdx := rwctxConsistentHashGlobal.GetNodeIdx(h, unhealthyIdx)
```

Comment:
Nice catch, this change must fix previous version issue with sharding skew.

Reply:
This issue was a part of current PR first impementation. 

 Hash was calculated from healthy nodes, instead of global nodes list.